### PR TITLE
Added location prefixes for lib, mod, doc, etc.

### PR DESCRIPTION
**Pull request by bekozi**
Ping @rokuingh.

### CODE REVIEW

**Comment by ocefpaf** [comment]
@bekozi do you have merge rights?

@rokuingh do you want to be added as a recipe maintainer? If you do @bekozi can your GitHub handle [here](https://github.com/conda-forge/esmf-feedstock/blob/master/recipe/meta.yaml#L28-L31) in this PR.

File: recipe/build.sh [hunk 1]
```diff
@@ -6,6 +6,11 @@ fi
 
 export ESMF_DIR=$(pwd)
 export ESMF_INSTALL_PREFIX=${PREFIX}
+export ESMF_INSTALL_BINDIR=${PREFIX}/bin
+export ESMF_INSTALL_DOCDIR=${PREFIX}/doc
+export ESMF_INSTALL_HEADERDIR=${PREFIX}/include
+export ESMF_INSTALL_LIBDIR=${PREFIX}/lib
+export ESMF_INSTALL_MODDIR=${PREFIX}/mod
```
Comment (ocefpaf) [inline recipe/build.sh]: Much better than my ugly hack!  Thanks @bekozi!

Comment (bekozi) [inline recipe/build.sh]: I should have gotten these in the recipe sooner. :flushed:

**Comment by bekozi** [comment]
> @bekozi do you have merge rights?

I believe so. I'm guessing all check should be green before merging... I admit, I am a bit skittish with the new system. I'll add @rokuingh as a maintainer.

**Comment by ocefpaf** [comment]
> I admit, I am a bit skittish

Don't be!  It is a new car, but we have insurance :wink:

**Comment by ocefpaf** [comment]
@bekozi can you bump the build [number](https://github.com/conda-forge/esmf-feedstock/pull/2/files#diff-e178b687b10a71a3348107ae3154e44cR10)? Also, if you add `[skip appveyor]` in your commit message AppVeyor won't start (and the conda-forge queue won't increase unnecessarily since we are skipping Windows anyway.)

**Comment by pelson** [comment]
> @bekozi can you bump the build number?

Let's take a step back a moment. When we had an all in one build system (like conda-recipes-scitools and ioos/conda-recipes) we needed to bump the version *whenever* we made the slightest change to the recipe so that we could confirm that the recipe still functioned. Now however, even if we make non-functional changes (such as this one) the CI system builds the recipe (one of the main advantages of having dedicated CI for each recipe) so we know that this change *does work*. The only reason to bump the build number therefore is when we have functional changes, *or we really want the changes to be reflected in the recipe that is installed along with the conda-forge package*. 

For me, the canonical source of *useful* recipes is the feedstock, not the recipe which gets put on my system when I install a forge package. It *is* important that I can see *how* my installed package was installed, but it isn't important that it is up-to-date (e.g. maintainers and other non-functional items).

In short, I don't believe the build number *needs* bumping - for me, non-functional changes *never* need a new build.

**Comment by ocefpaf** [comment]
> In short, I don't believe the build number needs bumping - for me, non-functional changes never need a new build.

I understand your point. But I do believe the changes in here deserve a new build number. If I got ESMF correctly it saves a makefile with those vars and ESMPY read that file.  So there is a functional change downstream. (Also, I am having trouble building nco with ESMF and I believe that this change will help.  More in this later today.)

**Comment by bekozi** [comment]
I bumped the build number. @pelson's point is well-taken.

**Comment by ocefpaf** [comment]
I canceled AppVeyor manually. (I guess we cannot use spaces within the brackets in [skip appveyor].)

**Comment by bekozi** [comment]
Oops...thanks.

**Comment by pelson** [comment]
Thanks @bekozi and @ocefpaf. I think it is completely right to bump the build number on a per-case basis. :+1:
